src/visitors/defuse_analyze_visitor.cpp

```diff
@@ -418,9 +426,18 @@ DUChain DefUseAnalyzeVisitor::analyze(const ast::Ast& node, const std::string& n
     visiting_lhs = false;
     current_symtab = global_symtab;
     unsupported_node = false;
+    auto symbol = current_symtab->lookup_in_scope(variable_name);
+    // If symbol is local_var or is not present on the current_symtab (in this case same as the
+    // global symtab) then we only look for evaluation of the DefUse chain as local var
+    if ((symbol != nullptr && symbol->has_any_property(NmodlType::local_var)) ||
+        symbol == nullptr) {
```

Comment:
as discussed, we can remove `|| symbol == nullptr` this?

Reply:
I figured out that this cannot be removed. Since we check the `current_symtab`, which in this case is the `global_symtab`, we can only check whether the variable is a global one or not. I have changed a bit this check, so let me know if the new version improves this.